opentreemap/importer/models/species.py

```diff
@@ -263,6 +263,7 @@ def validate_itree_code_field(self):
         else:
             self.cleaned[fields.species.ITREE_PAIRS] = pairs
 
+    @transaction.atomic
```

Comment:
Since this function is not fast, perhaps the transaction should just go around the `save`s at the end.